### ISSUE TITLE
Cypress 14: Breaking changes

### ISSUE THREAD

**Issue by jennifer-shehane** [cypress-io/cypress#29925]
### What would you like?

This issue is to track breaking changes planned for a future Cypress 14 release. 

### Planned breaking changes
- Electron update (33+)
- Remove support for Node.js 16 and 21: https://github.com/cypress-io/cypress/issues/29601
	- Including removing support for glibc under 2.28: https://github.com/cypress-io/cypress/issues/29930
- Move away from using document.domain to adhere to better security practices: https://github.com/cypress-io/cypress/issues/29590

- Remove support for older Chrome versions: https://github.com/cypress-io/cypress/issues/29977
- Remove support for older Firefox versions
- Remove support for `Cypress.backend('firefox:force:gc')` https://github.com/cypress-io/cypress/issues/30222
- Make JIT Compile default behavior for CT tests https://github.com/cypress-io/cypress/issues/30234
	- go experimentalJITCompile -> JITCompile, default it to true and mark it as deprecated
- Fix `before` hook running twice. https://github.com/cypress-io/cypress/issues/17940
- Remove patch jQuery 3.4.1 logic around width/height calculations: https://github.com/cypress-io/cypress/issues/29846
- Major version welcome screen - is this necessary? Will this screen work ok if this content is missing? Need to investigate
- Visibility changes:
	- https://github.com/cypress-io/cypress/pull/29778
	- https://github.com/cypress-io/cypress/pull/29741
	- https://github.com/cypress-io/cypress/pull/29688
	- https://github.com/cypress-io/cypress/pull/29689
	- https://github.com/cypress-io/cypress/pull/29680 
- Remove 'legacy' <v9 documentation. (example: https://docs.cypress.io/guides/references/legacy-configuration)

#### Bundler support removals
- Remove support for webpack-dev-server v3 inside @cypress/webpack-dev-server https://github.com/cypress-io/cypress/issues/29308
- Remove support for vite 3 inside @cypress/vite-dev-server: https://github.com/cypress-io/cypress/issues/29378
- Remove support for vite 2 inside @cypress/vite-dev-server: https://github.com/cypress-io/cypress/issues/29377

##### Vue related
- Remove support for Nuxt 2 (alpha) https://github.com/cypress-io/cypress/issues/30468
- Remove support for Vue 2 https://github.com/cypress-io/cypress/issues/30295

##### React related
- Remove support for Next.js =<13: https://github.com/cypress-io/cypress/issues/29583
- Remove support for React =<17 and make `react18` the new `react` testing harness: https://github.com/cypress-io/cypress/issues/29607 (blocked by https://github.com/cypress-io/cypress/issues/30511)
- remove deprecated `unmount` function https://github.com/cypress-io/cypress/issues/30623

##### Angular Related
- Remove support for Angular <17.2.0: https://github.com/cypress-io/cypress/issues/29582
- Support angular signals in cypress/angular: https://github.com/cypress-io/cypress/issues/29606

##### Other
- Remove support for Svelte 3 (alpha) https://github.com/cypress-io/cypress/issues/30492
- Remove support for Svelte 4 (alpha) https://github.com/cypress-io/cypress/issues/30692

#### Other Tooling removals
- Remove support for TypeScript 3 https://github.com/cypress-io/cypress/issues/30493
- Remove support for Create React App (CRA) https://github.com/cypress-io/cypress/issues/30028
- Remove support for `@vue/cli-service` https://github.com/cypress-io/cypress/issues/30481

### Breaking changes not in scope
- Fix Cypress.env not being reset on rerun: https://github.com/cypress-io/cypress/issues/6010
- Correct cy.screenshot's fullscreen behavior: https://github.com/cypress-io/cypress/pull/28930
- Update webpack-dev-server to v5: https://github.com/cypress-io/cypress/pull/28926 (done in https://github.com/cypress-io/cypress/issues/30347 which is in scope for cypress 14)

### Features (Currently planned for initial Cypress 14 release but can release after initial Cypress 14 and need a spike)
* Support Svelte 5 https://github.com/cypress-io/cypress/issues/29641
* Support Angular 19 https://github.com/cypress-io/cypress/issues/30175
* Support Next.js 15 https://github.com/cypress-io/cypress/issues/30445
* Support React 19 https://github.com/cypress-io/cypress/issues/29470
* Support Vite 6 https://github.com/cypress-io/cypress/issues/30591

### Why is this needed?

Mostly this release's aim is to reduce the maintenance load on the current team with supporting older versions of dependencies.

**Comment by MikeMcC399** [comment]
@jennifer-shehane / @AtofStryker 

- I added https://github.com/cypress-io/cypress/issues/29930 for breaking change consideration.

**Comment by MikeMcC399** [comment]
- Suggest to review https://github.com/cypress-io/cypress-documentation/issues/5446 for minimum Chrome version. The download source referred to by the following purges older versions. For instance 107.0.5304.121-1, released on Nov 24, 2022 was no longer available. Google itself only supports the current stable release and accepts bug reports for beta / dev / canary.

    https://github.com/cypress-io/cypress-docker-images/blob/master/factory/.env
    https://www.ubuntuupdates.org/package/google_chrome/stable/main/base/google-chrome-stable

**Comment by jennifer-shehane** [comment]
We could include a higher minimum browser version in this release to suggest people use. There wouldn't be any actual code changes to accompany that.

**Comment by MikeMcC399** [comment]
@jennifer-shehane 

Since older Chrome versions are still available through existing Cypress Docker images `cypress/browsers:node13.6.0-chrome-80-ff72` they could still be tested.

I'd suggest just bumping the Chrome minimum to `80`, in line with your https://github.com/cypress-io/cypress-documentation/issues/5446#issuecomment-2258525335 which would be a code change.

https://github.com/cypress-io/cypress/blob/40de87d26017dda24c680e6a16093fb65ee5a0c6/packages/launcher/lib/known-browsers.ts#L3-L10

**Comment by senpl** [comment]
It would be nice to push browser visibility API into this as this might be breaking change for some users visibility checks.

**Comment by jennifer-shehane** [comment]
@senpl Right, much of those changes are bordering on breaking changes that we could evaluate more freely with a breaking change release. Added it to list of consideration.

**Comment by MikeMcC399** [comment]
- https://github.com/cypress-io/cypress/issues/29977 added as separate issue for consideration

**Comment by MikeMcC399** [comment]
- https://github.com/cypress-io/cypress/issues/30028 should also be considered for Cypress 14 since CRA is no longer being supported.

**Comment by jennifer-shehane** [comment]
Closing as v14 branch has been merged into develop
